tests/unit/components/DocumentationTopic/DocumentationNav/VersionToggle.spec.js

```diff
@@ -0,0 +1,215 @@
+    wrapper.setProps({
+      versionList: ['V3', 'V2', 'V1'],
+    });
+
+    expect(wrapper.find('#version-toggle').element.value).toBe('V3');
+
+    wrapper.findAll('#version-toggle option').at(2).setSelected();
+
+    const options = wrapper.find('#version-toggle').findAll('option');
+
+    await options.at(2).setSelected();
+
+    expect(wrapper.find('#version-toggle').element.value).toBe('V1');
+    expect(mocks.$router.push).toHaveBeenCalledWith({ path: '/documentation/foo', query: { version: 'V1' } });
+    expect(provide.store.setPreferredVersion).toHaveBeenCalledWith('V1');
+  });
+
+  it('clears out the language query if version is the most recent version', async () => {
+    wrapper.setProps({
+      versionList: ['V3', 'V2', 'V1'],
+    });
+    await flushPromises();
+    expect(wrapper.find('#version-toggle').element.value).toBe('V3');
+
+    wrapper.findAll('#version-toggle option').at(0).setSelected();
+
+    const options = wrapper.find('#version-toggle').findAll('option');
+
+    await options.at(0).setSelected();
+
+    expect(mocks.$router.push).toHaveBeenCalledWith({ path: '/documentation/foo', query: { version: undefined } });
+
+    // const link = wrapper.find('.version-list-container').find('a.nav-menu-link');
+    // expect(link.props('to').query).toEqual({ version: undefined });
+  });
+
+  it('keeps extra query parameters when changing version', async () => {
+    const query = {
+      foo: 'foo',
+      bar: 'bar',
+    };
+    const postVersionQuery = {
+      foo: 'foo',
+      bar: 'bar',
+      version: 'V1',
+    };
+    const mocksWithQuery = {
+      ...mocks,
+      $route: {
+        path: '/documentation/foo',
+        query,
+      },
+    };
+    wrapper = createWrapper(undefined, mocksWithQuery);
+    wrapper.setProps({
+      versionList: ['V3', 'V2', 'V1'],
+    });
+
+    wrapper.findAll('#version-toggle option').at(2).setSelected();
+
+    const options = wrapper.find('#version-toggle').findAll('option');
+
+    await options.at(2).setSelected();
+
+    expect(wrapper.find('#version-toggle').element.value).toBe('V1');
+    expect(mocks.$router.push).toHaveBeenCalledWith({ path: '/documentation/foo', query: postVersionQuery });
+    // expect(mocks.$route).toEqual((query));
+  });
+
+  it('changes the model, if the preferred version changes and is within the versionlist for the page', async () => {
+    wrapper = shallowMount(VersionToggle, {
+      mocks,
+      propsData,
+      provide: {
+        store: DocumentationTopicStore,
+      },
+      stubs,
+    });
+
+    DocumentationTopicStore.setPreferredVersion('V2');
+    expect(DocumentationTopicStore.state.preferredVersion).toBe('V2');
+
+    await flushPromises();
+    wrapper.setProps({
+      versionList: ['V3', 'V2', 'V1'],
+    });
+    expect(wrapper.find('#version-toggle').element.value).toBe('V2');
+  });
+
+  it('does not change the model, if the preferred version changes and is not within the versionlist for the page', async () => {
```

Comment:
I think this test is a bit misleading. 

The logic in `currentVersion` is it returns the preferredVersion if its part of the `versionList` OR it returns `versionList[0]`, but if you say have picked `versionList[1]` it would change model to `versionList[0]` :)